crates/router/src/connector/braintree/braintree_graphql_transformers.rs

```diff
@@ -105,36 +172,42 @@ impl TryFrom<&BraintreeRouterData<&types::PaymentsAuthorizeRouterData>>
             item.router_data.request.currency,
             Some(metadata.merchant_config_currency),
         )?;
-
-        match item.router_data.request.payment_method_data.clone() {
-            domain::PaymentMethodData::Card(_) => {
-                if item.router_data.is_three_ds() {
-                    Ok(Self::CardThreeDs(BraintreeClientTokenRequest::try_from(
-                        metadata,
-                    )?))
-                } else {
-                    Ok(Self::Card(CardPaymentRequest::try_from((item, metadata))?))
+        match item.router_data.request.connector_mandate_id() {
+            Some(connector_mandate_id) => Ok(Self::Mandate(MandatePaymentRequest::try_from((
+                item,
+                connector_mandate_id,
+                metadata,
+            ))?)),
+            None => match item.router_data.request.payment_method_data.clone() {
+                domain::PaymentMethodData::Card(_) => {
+                    if item.router_data.is_three_ds() {
+                        Ok(Self::CardThreeDs(BraintreeClientTokenRequest::try_from(
+                            metadata,
+                        )?))
+                    } else {
+                        Ok(Self::Card(CardPaymentRequest::try_from((item, metadata))?))
+                    }
                 }
-            }
-            domain::PaymentMethodData::CardRedirect(_)
-            | domain::PaymentMethodData::Wallet(_)
-            | domain::PaymentMethodData::PayLater(_)
-            | domain::PaymentMethodData::BankRedirect(_)
-            | domain::PaymentMethodData::BankDebit(_)
-            | domain::PaymentMethodData::BankTransfer(_)
-            | domain::PaymentMethodData::Crypto(_)
-            | domain::PaymentMethodData::MandatePayment
-            | domain::PaymentMethodData::Reward
-            | domain::PaymentMethodData::RealTimePayment(_)
-            | domain::PaymentMethodData::Upi(_)
-            | domain::PaymentMethodData::Voucher(_)
-            | domain::PaymentMethodData::GiftCard(_)
-            | domain::PaymentMethodData::CardToken(_) => {
-                Err(errors::ConnectorError::NotImplemented(
-                    utils::get_unimplemented_payment_method_error_message("braintree"),
-                )
-                .into())
-            }
+                domain::PaymentMethodData::CardRedirect(_)
+                | domain::PaymentMethodData::Wallet(_)
+                | domain::PaymentMethodData::PayLater(_)
+                | domain::PaymentMethodData::BankRedirect(_)
+                | domain::PaymentMethodData::BankDebit(_)
+                | domain::PaymentMethodData::BankTransfer(_)
+                | domain::PaymentMethodData::Crypto(_)
+                | domain::PaymentMethodData::MandatePayment
```

Comment:
Define the request for MandatePayment as payment_method, refer to `cybersource/transformer.rs`

Reply:
Fixed